Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java

```diff
@@ -0,0 +1,215 @@
+        }
+
+        String releaseDate = (String) payload.get("releaseDate");
+        if (releaseDate == null) {
+            throw new MissingAttributeException("Request body must contain 'releaseDate'");
+        }
+
+        String coverImage = (String) payload.get("coverImage");
+        if (coverImage == null) {
+            throw new MissingAttributeException("Request body must contain 'coverImage'");
+        }
+
+        Boolean isActive = true;
+
+        Listing listing = new Listing(
+                sellerUserId.longValue(),
+                isbn,
+                title,
+                floatPrice,
+                author,
+                publisher,
+                description,
+                inventory,
+                releaseDate,
+                isActive,
+                coverImage
+        );
+
+        repo.save(listing);
+
+        return listing;
+    }
+
+    @PutMapping("/listings")
+    public Map<String, Object> putListing(@RequestBody Map<String, Object> payload) {
+        Map<String, Object> map = new HashMap<>();
+
+        Integer listingId = (Integer) payload.get("listingId");
+        if (listingId == null) {
+            throw new MissingAttributeException("Request body must contain 'listingId'");
+        }
+
+        Listing listing = repo.findByListingId(listingId.longValue());
+
+        if (listing == null) {
+            throw new NotFoundException("Listing with id [" + listingId + "] cannot be found");
+        }
+
+        String isbn = (String) payload.get("isbn");
+        if (isbn != null) {
+            listing.setISBN(isbn);
+            map.put("isbn", isbn);
+        }
+        String title = (String) payload.get("title");
+        if (title != null) {
+            listing.setTitle(title);
+            map.put("title", title);
+        }
+
+        String price = (String) payload.get("price");
+        if (price != null) {
+            try {
+                Float floatPrice = Float.valueOf(price);
+                listing.setPrice(floatPrice);
+                map.put("price", price);
+            } catch (NumberFormatException e) {
+                throw new MissingAttributeException("Attribute 'price' must be formatted in decimal as a string (ex. '1.5')");
+            }
+            listing.setPrice(Float.valueOf(price));
+        }
+
+        String author = (String) payload.get("author");
+        if (author != null) {
+            listing.setAuthor(author);
+            map.put("author", author);
+        }
+        String publisher = (String) payload.get("publisher");
+        if (publisher != null) {
+            listing.setPublisher(publisher);
+            map.put("publisher", publisher);
+        }
+        String description = (String) payload.get("description");
+        if (description != null) {
+            listing.setDescription(description);
+            map.put("description", description);
+        }
+        Integer inventory = (Integer) payload.get("inventory");
```

Comment:
Add a check to ensure that inventory is never negative. same goes for the put request

Reply:
Add negative value checks for inventory and price for POST and PUT.